dbms/src/Common/MemoryTracker.cpp

```diff
@@ -111,7 +117,11 @@ void MemoryTracker::alloc(Int64 size, bool check_memory_limit)
     reportAmount();
 
     if (!next.load(std::memory_order_relaxed))
+    {
         CurrentMetrics::add(metric, size);
+        if (shared_column_data_mem_tracker)
```

Comment:
Maybe data race here?

Reply:
Yes, I aggree, but theoretically there will be data race here, not sure if our ASan tests will complain it or not. I'm ok with current implementation, and I think we can fix it if it breaks the ASan tests.